posix: shm: build error with minimal libc

**Describe the bug**

In `lib/posix/options/shm.c` we don't include `stdio.h` header but use standard `SEEK_xxx` definitions (i.e. `SEEK_SET`) which cause build issues (if the minimal libc is used). I.e.: (tests/posix/shm/portability.posix.shm test build):

```
/verification_workspace/zephyr/lib/posix/options/shm.c:144:7: error: use of undeclared identifier 'SEEK_SET'
        case SEEK_SET:
             ^
/verification_workspace/zephyr/lib/posix/options/shm.c:147:7: error: use of undeclared identifier 'SEEK_CUR'
        case SEEK_CUR:
             ^
/verification_workspace/zephyr/lib/posix/options/shm.c:150:7: error: use of undeclared identifier 'SEEK_END'
        case SEEK_END:
             ^
3 errors generated.
```

**Expected behavior**
No build error should happen

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: MWDT 2024.06
 - Commit SHA: recent main, 1ed04e7b49c5bec8e09f81480b7f95ae51b59b10

--- a/lib/posix/options/shm.c
+++ b/lib/posix/options/shm.c
@@ -5,6 +5,7 @@
  */
 
 #include <string.h>
+#include <stdio.h>
 
 #include <kernel_arch_interface.h>
 #include <zephyr/kernel.h>

